src/index.js

```diff
@@ -224,6 +224,28 @@ module.exports = (
     .then(finalizeHandler);
   }
   else {
+    if (typeof watch === 'object') {
+      if (!watch.watch)
+        throw new Error('Watcher class must be a valid Webpack WatchFileSystem class instance (https://github.com/webpack/webpack/blob/master/lib/node/NodeWatchFileSystem.js)');
+      compiler.watchFileSystem = watch;
+      watch.inputFileSystem = compiler.inputFileSystem;
+    }
+    else {
+      // Used for inspecting Webpack watcher API
```

Comment:
I don't understand this part. This else is for `typeof watch !== 'object'`? Is the assumption that `watch` is `true` at this point?

Reply:
Exactly - `watch: true` is the normal API, while `watch: new Watcher()` is the hooked API version.